From 941862202864ed2f65018b5af3f8b5d340ba5116 Mon Sep 17 00:00:00 2001
From: Dan Prince <dprince@redhat.com>
Date: Thu, 16 May 2013 13:54:56 -0400
Subject: [PATCH] Don't hard code libvirt emulator.

Updates configure-bootstrap-vm so that it determines the correct
location of the emulator binary and injects it as a parameter
into the libvirt template.
---
 bootstrap/bootstrap.xml          | 2 +-
 bootstrap/configure-bootstrap-vm | 5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/bootstrap/bootstrap.xml b/bootstrap/bootstrap.xml
index 726022e..1425e7e 100644
--- a/bootstrap/bootstrap.xml
+++ b/bootstrap/bootstrap.xml
@@ -16,7 +16,7 @@
   <on_reboot>restart</on_reboot>
   <on_crash>restart</on_crash>
   <devices>
-    <emulator>/usr/bin/kvm</emulator>
+    <emulator>%(emulator)s</emulator>
     <disk type='file' device='disk'>
       <driver name='qemu' type='qcow2' cache='writeback'/>
       <source file='%(imagefile)s'/>
diff --git a/bootstrap/configure-bootstrap-vm b/bootstrap/configure-bootstrap-vm
index 665fbf2..97d6187 100755
--- a/bootstrap/configure-bootstrap-vm
+++ b/bootstrap/configure-bootstrap-vm
@@ -34,6 +34,11 @@ def main():
     if args.image is not None:
         params['imagefile'] = args.image
 
+    if os.path.exists("/usr/bin/kvm"): # Debian
+        params['emulator'] = "/usr/bin/kvm"
+    elif os.path.exists("/usr/bin/qemu-kvm"): # Redhat
+        params['emulator'] = "/usr/bin/qemu-kvm"
+
     libvirt_template = source_template % params
     conn=libvirt.open("qemu:///system")
     a = conn.defineXML(libvirt_template)
-- 
1.8.1.6

